analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd

```diff
@@ -0,0 +1,333 @@
+histologies_rna <- read_tsv(metadata_file, guess_max = 10000) %>%
+  filter(composition=="Solid Tissue",
+                  tumor_descriptor == "Initial CNS Tumor",
+                  experimental_strategy == "RNA-Seq", 
+                  RNA_library == "stranded") %>%
+  rename(Kids_First_Biospecimen_ID_RNA = Kids_First_Biospecimen_ID) %>% 
+  select(Kids_First_Participant_ID, Kids_First_Biospecimen_ID_RNA, cancer_group, OS_status, OS_days, PFS_days) %>%
+  distinct() 
+
+# read in tp53 scores
+tp53_scores <- read_tsv(tp53_file) %>%
+  select(Kids_First_Biospecimen_ID_RNA, sample_id, tp53_score) %>%
+  distinct()
+
+# read in telomerase scores
+tel_scores <- read_tsv(tel_stranded_file) %>%
+  rename(Kids_First_Biospecimen_ID_RNA = SampleID,
+                telomerase_score = NormEXTENDScores) %>%
+  arrange(Kids_First_Biospecimen_ID_RNA)
+
+
+# combine tp53 scores and telomerase score to the dataframe
+# note that we do not need to recode categories since the survival_models.R functions will take care of it
+meta_indep <- histologies_rna %>%
+  # add tp53 and telomerase scores by matching Kids_First_Biospecimen_ID_RNA
+  left_join(tp53_scores, by = "Kids_First_Biospecimen_ID_RNA") %>%
+  left_join(tel_scores, by = "Kids_First_Biospecimen_ID_RNA") %>% 
+  arrange(Kids_First_Biospecimen_ID_RNA) %>%
+  distinct(Kids_First_Participant_ID, .keep_all=TRUE) %>%
+  # remove samples without survival data
+  drop_na(OS_days) %>%
+  # convert to years
+  mutate(OS_years = OS_days / 365.25)
+
+# Show the data we will use for analysis
+meta_indep
+```
+ 
+#### Create subsets of data for different models
+
+We create two new tibbles from `meta_indep`:
+
++ `meta_indep_hgg` which contains a new column `hgg_group` that contains levels (in order) `non-HGAT` and `HGAT` indicating whether it's an HGG sample
++ `meta_indep_cancers` which is subsetted to only certain cancer groups of interest that have suitable sample sizes which we find here.
+
+```{r hgg_split}
+# Define HGG groups
+hgg_cancer_groups <- c("High-grade glioma astrocytoma", "Diffuse midline glioma", "Diffuse intrinsic pontine glioma") 
+hgg_levels <- c("non-HGAT", "HGAT")
+
+# separate into HGG or non-HGG and make the new column a factor
+meta_indep_hgg <- meta_indep %>% 
+   mutate(hgg_group = ifelse(cancer_group %in% hgg_cancer_groups,
+                             hgg_levels[2], # second is HGAT 
+                             hgg_levels[1]), # first is non-HGAT
+          hgg_group = factor(hgg_group, levels = hgg_levels) # ensures non-HGAT is the reference level
+   )
+```
+
+```{r find_cancer_groups_to_use}
+# Define cancer groups of interest with higher Ns
+# We want to only include groups that have at least THREE in the DECEASED category. 
+# Find them here:
+cancer_groups_of_interest_df <- meta_indep %>%
+  drop_na() %>%
+  count(cancer_group,OS_status) %>%
+  spread(OS_status, n,
+         # any missing values (indicating 0 deceased or 0 living) should be 0, not NA
```

Comment:
Is this comment correct? I am confused.

Reply:
The comment is explaining line 147 - `fill = 0`

Reply:
This part of the code is still counting the number of events, and `NA`s will occur when spreading if there is no recorded event. `indicating 0 deceased or 0 living` is referring to number of participants, not the recoding.

I'll make sure this is doing what I think it is and rephrase a little.

Reply:
My issue is with the phrasing. Using "no events" seems less confusing to me.